consensus/src/marshal/ingress/mailbox.rs

```diff
@@ -238,6 +245,21 @@ impl<S: Scheme, B: Block> Mailbox<S, B> {
         rx
     }
 
+    /// Returns an [AncestorStream] over the ancestry of a given block, leading up to the end bound.
+    ///
+    /// If the starting block is not found, `None` is returned.
+    pub async fn ancestry(
```

Comment:
If ancestry was built-in to `Application`, I "think" you wouldn't need to know about Marshal anymore?

Reply:
Yeah - I changed this to remove the bound from `AncestorStream` and just pass it into the `Application::build` function.

Because it's a lazy-populated stream, the application can choose how far back they want to go.